### DISCUSSION TITLE
Can I use openapi.Response Model?

### DISCUSSION THREAD

**Discussion by ehdgua01** [fastapi/fastapi#4093] [Questions] [answered]
Can I use `fastapi.openapi.Response` model for describing my APIs? (for type hint)
Isn't it an experimental model?

Example

```python
from fastapi.openapi.models import Response

router = APIRouter()

@router.get(
    "",
    responses: {
       200: Response(description="Some text").dict(),
    },
)
def view():
    ...
```

**Comment by fabioueno** [comment]
Hey, @ehdgua01! How are you? 🙂

Extremely late reply here, but I hope this still has some value. 😬

> Isn't it an experimental model?

Not at all! In fact, [there's a place and time for using it](https://fastapi.tiangolo.com/tutorial/response-model/#return-a-response-directly). As the [advanced docs](https://fastapi.tiangolo.com/advanced/response-directly) state, the most interesting case is if you need/want more flexibility in your return. Say you don't need to validate your data, or don't know what fields you have at hand and just want to return a JSON, that's the perfect use case. 😉

By using `fastapi.Response`, you still get editor support and stuff but get enough flexibility in your responses. 🥳

Hope that helps!

**Reply by YuriiMotov** [reply]
This is completely different `Response` class. The question was about `from fastapi.openapi.models import Response` that is Pydantic model

**Comment by YuriiMotov** [comment, marked as answer]
I think it's safe to use this model this way.
FastAPI internally uses it to validate the data you pass to `requests` parameter
